stage1/init/common/pod.go

```diff
@@ -1105,3 +1112,139 @@ func protectSystemFiles(opts []*unit.UnitOption, appName types.ACName) []*unit.U
+					return opts, nil
+				}
+			}
+			seccompErrno = string(seccomp.Errno())
+		}
+	}
+
+	// rkt default whitelist
+	if len(seccompSet) == 0 {
+		charMode = ""
+		seccompSet = RktDefaultSeccompWhitelist
+	}
+
+	// Append options
+	if seccompErrno != "" {
+		opts = append(opts, unit.NewUnitOption("Service", "SystemCallErrorNumber", seccompErrno))
+	}
+	seccompStr := ""
+	for i, s := range seccompSet {
+		next := i + 1
+		seccompStr += (s + " ")
+		if next%50 == 0 || next == len(seccompSet) {
+			opts = append(opts, unit.NewUnitOption("Service", "SystemCallFilter", fmt.Sprintf("%s%s", charMode, seccompStr)))
+			seccompStr = ""
+		}
+	}
+
+	return opts, nil
+}
+
+// parseLinuxSeccompSet gets an appc seccomp set and returns an array
+// of values suitable for systemd SystemCallFilter.
+func parseLinuxSeccompSet(p *stage1commontypes.Pod, s types.LinuxSeccompSet) (syscallFilter []string, flag string, err error) {
+	for _, item := range s.Set() {
+		if item[0] == '@' {
+			// Wildcards
+			wildcard := strings.SplitN(string(item), "/", 2)
+			if len(wildcard) != 2 {
+				continue
+			}
+			scope := wildcard[0]
+			name := wildcard[1]
+			switch scope {
+			case "@appc":
+				// appc-reserved wildcards
+				switch name {
+				case "all":
+					return nil, "all", nil
+				case "empty":
+					return nil, "empty", nil
+				}
+			case "@docker":
+				// Docker-originated wildcards
+				switch name {
+				case "default-blacklist":
+					syscallFilter = append(syscallFilter, DockerDefaultSeccompBlacklist...)
+				case "default-whitelist":
+					syscallFilter = append(syscallFilter, DockerDefaultSeccompWhitelist...)
+				}
+			case "@rkt":
+				// Custom rkt wildcards
+				switch name {
+				case "default-blacklist":
+					syscallFilter = append(syscallFilter, RktDefaultSeccompBlacklist...)
+				case "default-whitelist":
+					syscallFilter = append(syscallFilter, RktDefaultSeccompWhitelist...)
+				}
+			case "@systemd":
+				// Custom systemd wildcards (systemd >= 231)
+				_, systemdVersion, err := GetFlavor(p)
+				if err != nil || systemdVersion < 231 {
+					return nil, "", errors.New("Unsupported or unknown systemd version, needed systemd >= v231")
+				}
+				switch name {
+				case "clock":
+					syscallFilter = append(syscallFilter, "@clock")
+				case "default-whitelist":
+					syscallFilter = append(syscallFilter, "@default")
+				case "mount":
+					syscallFilter = append(syscallFilter, "@mount")
+				case "network-io":
+					syscallFilter = append(syscallFilter, "@network-io")
+				case "obsolete":
+					syscallFilter = append(syscallFilter, "@obsolete")
+				case "privileged":
+					syscallFilter = append(syscallFilter, "@privileged")
+				case "process":
+					syscallFilter = append(syscallFilter, "@process")
+				case "raw-io":
+					syscallFilter = append(syscallFilter, "@raw-io")
+				}
+			}
+		} else {
+			// Plain syscall name
+			syscallFilter = append(syscallFilter, string(item))
+		}
+	}
+	return syscallFilter, "", nil
+}
```

Comment:
should parseLinuxSeccompSet and getSeccompFilter be moved to their own source file (stage1/init/common//seccomp.go)?